src/admin/components/Fields/ColoredChipFieldArray.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from "classnames";
+import { ArrayField, ArrayFieldProps, ChipField, FunctionField, SingleFieldList, useRecordContext } from "react-admin";
+
+import { Choice } from "@/admin/types/common";
+import { useGetV2SitesSitePolygon } from "@/generated/apiComponents";
+
+interface ColoredChipFieldArrayProps extends Omit<ArrayFieldProps, "children"> {
+  choices: Choice[];
+}
+
+const POLYGON_SUBMITTED_TYPE_CLASSNAME_MAP: { [key: string]: string } = {
+  Approved: "!bg-green-30 tag-approved-color",
+  Submitted: "!bg-blue-200 tag-submitted-color",
+  Draft: "!bg-grey-200 tag-draft-color",
+  "Needs info": "!bg-tertiary-50 tag-need-info-color",
+  Unknown: ""
+};
+
+const ColoredChipFieldArray = (props: ColoredChipFieldArrayProps) => {
+  const recordContext = useRecordContext();
+  const { data: getPolygonsToSite } = useGetV2SitesSitePolygon({ pathParams: { site: recordContext.uuid } });
+  //fix: RA crashes when null or undefined passed to an arrayField
+  if (!getPolygonsToSite?.length || !Array.isArray(getPolygonsToSite)) {
+    return (
+      <div className="text-14 w-fit-content whitespace-nowrap rounded-[3px] bg-grey-200 px-2 text-grey-500">
+        {props.emptyText || "No Polygons"}
+      </div>
+    );
+  }
+
+  function groupPolygonsByStatus(polygons: any[]) {
```

Comment:
This function doesn't appear to depend on any props to the component - please move out of the component definition for clarity and performance.